app/scripts/metamask-controller.js

```diff
@@ -4728,25 +4760,21 @@ export default class MetamaskController extends EventEmitter {
     // append selectedNetworkClientId to each request
     engine.push(createSelectedNetworkMiddleware(this.controllerMessenger));
 
-    const { selectedNetworkClientId } = this.networkController.state;
-
-    const selectedNetworkClientIdForDomain =
-      this.selectedNetworkController.getNetworkClientIdForDomain(origin);
-
-    // Not sure that this will happen anymore
-    if (selectedNetworkClientIdForDomain === undefined) {
-      this.selectedNetworkController.setNetworkClientIdForDomain(
-        origin,
-        selectedNetworkClientId,
-      );
-    }
-
-    let proxyProviderForDomain = provider;
-    if (this.preferencesController.getUseRequestQueue()) {
-      proxyProviderForDomain =
-        this.selectedNetworkController.getProviderAndBlockTracker(
-          origin,
-        ).provider;
+    // Use proxyClient from selectedNetworkController when:
+    // 1. feature flag for perDomainNetwork is on
+    // 2  the origin is in the selectedNetworkController's `domains` state
```

Comment:
I think it may be more helpful to future devs to explain why we prefer using the global as opposed to setting the networkClient for the origin if it doesn't exist already and then using that. I could see someone trying to set a default here and not having enough background/context to understand why we don't want to do that

Reply:
Chained the slant of the comments here: https://github.com/MetaMask/metamask-extension/pull/23130/commits/f2c2dbdb049f2e7268be45d684c8714acc1833f9 . Let me know what you think